lib/mutant/ast/regexp.rb

```diff
@@ -3,8 +3,18 @@ module AST
     # Regexp source mapper
     module Regexp
       UNSUPPORTED_EXPRESSION_TYPE = :conditional
+
       private_constant(*constants(false))
 
+      # Parse regex string into expression
+      #
+      # @param regexp [String]
+      #
+      # @return [Regexp::Expression]
+      def self.parse(regexp)
+        ::Regexp::Parser.parse(regexp, "ruby/#{RUBY_VERSION.to_f}")
```

Comment:
This is not a good idea. You rely on transitive property `String -> Float -> OtherString`. Lets do something more explicit, splitting the string and joining together the first 2 components.

Using type coercions that only by "accident" produce a value mutation (effectively you mutate a string) is error prone.